addon/models/osf-model.js

```diff
@@ -65,11 +65,55 @@ export default DS.Model.extend(HasManyQuery.ModelMixin, {
                     remove: relation.canonicalMembers.list.filter(m => currentIds.indexOf(m.getRecord().get('id')) === -1)
                 };
 
-                var other = this.get('_dirtyRelationships.${rel}') || {};
+                var other = this.get(`_dirtyRelationships.${rel}`) || {};
                 Ember.merge(other, changes);
                 this.set(`_dirtyRelationships.${rel}`, other);
             }
         });
         return this._super(...arguments);
-    }
+    },
+
+    /*
+     * Query a hasMany relationship with query params
+     *
+     * @method queryHasMany
+     * @param {String} propertyName Name of a hasMany relationship on the model
+     * @param {Object} queryParams A hash to be serialized into the query string of the request
+     * @param {Object} [ajaxOptions] A hash of options to be passed to jQuery.ajax
+     * @returns {ArrayPromiseProxy} Promise-like array proxy, resolves to the records fetched
+     */
+    queryHasMany(propertyName, queryParams, ajaxOptions) {
+        const reference = this.hasMany(propertyName);
+        const promise = new Ember.RSVP.Promise((resolve, reject) => {
+            // HACK: ember-data discards/ignores the link if an object on the belongsTo side
+            // came first. In that case, grab the link where we expect it from OSF's API
+            const url = reference.link() || this.get(`links.relationships.${propertyName}.links.related.href`);
+            if (!url) {
+                reject(`Could not find a link for '${propertyName}' relationship`);
+                return;
+            }
+            const options = Ember.merge({
+                url,
+                data: queryParams,
+                headers: Ember.get(OSFAdapter, 'headers'),
+            }, ajaxOptions);
+
+            authenticatedAJAX(options).then(
+                Ember.run.bind(this, this.__queryHasManyDone, resolve),
+                reject
+            );
+        });
+
+        const ArrayPromiseProxy = Ember.ArrayProxy.extend(Ember.PromiseProxyMixin);
+        return ArrayPromiseProxy.create({ promise });
+    },
+
+    __queryHasManyDone(resolve, payload) {
```

Comment:
I like the double underscore prefix for "superprivate" methods that should never be called directly (only as a callback). AFAIK, this is not a convention in JS, but I like it.